### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by AlteredCoder**
The idea behind this refactor is to be able to load "dynamically" backend that can be different according to the OS the blocker is running

### CODE REVIEW

File: backend.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (b *backendCtx) Run(dbCTX *sqlite.Context, frequency time.Duration) error {
+	err := b.ctx.Run(dbCTX, frequency)
+	if err != nil {
+		log.Fatalf(err.Error())
```
Comment (buixor) [inline backend.go:32]: maybe return error instead of fatal'ing here

File: iptables.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+			if err := ipt.DeleteBan(ba); err != nil {
+				return err
+			}
+		}
+
+		bansToAdd, err := getNewBan(dbCTX)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Adding ban")
```
Comment (buixor) [inline iptables.go]: maybe display the numbers of ban that are going to be added, ie `Adding %d bans`

**Review by buixor — COMMENTED** [review]
LGTM, minor remarks

File: iptables.go (hunk 2 above)
Comment (AlteredCoder) [reply iptables.go]: this is a debug, it will be removed ^^

File: backend.go (hunk 1 above)
Comment (AlteredCoder) [reply backend.go:32]: i don't think we want to return an error here since this function is called in a go routine

**Pull request merged**